lib/Twig/Extension/Core.php

```diff
@@ -852,6 +852,34 @@ function twig_in_filter($value, $compare)
     return false;
 }
 
+/** 
+ * Returns a trimmed string.
+ * 
+ * @param string $string
+ * @param string $characterMask
+ * @param string $mode
+ *
+ * @return string
+ *
+ * @throws Twig_Error_Runtime When an invalid type is used (not 'left', 'right' or 'both')
+ */
+function twig_trim_filter($string, $characterMask = null, $type = 'both')
+{
+    if (null === $characterMask) {
+        $characterMask = " \t\n\r\0\x0B";
+    }
+    switch ($type) {
+        case 'left':
+            return ltrim($string, $characterMask);
+        case 'right':
+            return rtrim($string, $characterMask);
+        case 'both':
+            return trim($string, $characterMask);
+        default:
+            throw new Twig_Error_Runtime(sprintf('trim type must be "left", "right" or "both". Got %s.', $type));
```

Comment:
```php
throw new Twig_Error_Runtime(sprintf('trim type must be "left", "right" or "both", got "%s".', is_object($type) ? get_class($type) : (gettype($type).'#'.$type)));
```
(this will fail `resources` though)

On a side note, as `both` is prop. used the most, please make it the first check of the `switch`